Feature/review 업데이트

namsoo5 opened the pull request:
MVVM구조로 Rx를 구현해보려고 했습니다.
저번에 배웠던 내용들을 잘 모르겠어서 최대한 해보려고 했습니다.
안 좋은 코드이거나 부족한 부분이 있다면 말씀해주세요!!

- Adjust 이벤트 로깅 추가
- 업데이트를 위한 후기탭 제거
- 라이브러리 추가 (Kingfisher, Adjust)
- 이미지Asset 추가

namsoo5 committed:
UPDATE: 동아리 검색 Table구현

- 카테고리 Factory생성
- RX로 테이블뷰 생성

namsoo5 committed:
UPDATE: 검색 결과에 따른 테이블 기능 구현

- 데이터 ViewModel생성
- 데이터에따른 라벨과 버튼 hidden설정

namsoo5 committed:
UPDATE: 동아리 검색기능구현

- Cell Binding
- CellType: 교내동아리, 연합동아리 구분

namsoo5 committed:
UPDATE: Kingfisher추가, 이미지 로딩방식 변경

- 라이브러리를 이용한 캐싱처리

namsoo5 committed:
UPDATE: 키보드상 Done키 이벤트 설정

- 완료버튼 클릭시 통신되도록 구현

namsoo5 committed:
UPDATE: 후기탭 동아리 리스트 Paging처리 버그 수정

- 빈배열시 통신 안하도록 구현

namsoo5 committed:
UPDATE: 후기등록 뷰 구현

- 후기등록 버튼을 통한 동아리 정보입력뷰
- 동아리 정보 입력 로직 구현
- 등록되지 않은 동아리 입력
- 등록된 동아리 입력
- 별점 추천 구현중

namsoo5 committed:
UPDATE: 서버 수정에따른 모델 변경 및 탭바 Hidden

- 후기 등록시 보이는 Tabbar Hidden처리
- 동아리 리스트 통신모델 자료형 변경

namsoo5 committed:
UPDATE: 후기별점 기능 구현

- 별점 클릭에따른 별이미지 채우기
- 별점 모든 평가시 다음버튼 활성화

namsoo5 committed:
UPDATE: 별점평가후 리뷰쓰는 뷰컨구현 앱 Flow수정

- 간단후기 뷰컨트롤러 생성
- 기존 후기쓰기버튼을통해 push방식을 present방식으로 변경

namsoo5 committed:
UPDATE: 간단 후기 글작성구현

- 뷰 구현
- 기능 구현
- 터치시 키보드 숨기기

namsoo5 committed:
CREATE: Adjust 이벤트 추가

- facebook, adjust 이벤트 묶어서 함수화

namsoo5 committed:
UPDATE: 후기Tab 제거

- 앱 업데이트를 위한 아직 미구현인 탭부분 제거

namsoo5 committed:
FIX: Adjust 테스트용 -> 배포용으로 설정 변경

hyejuuu commented on SsgSag/Screens/Review/ViewModels/SimpleReviewViewModel.swift:
```diff
@@ -0,0 +1,41 @@
+    
+    func checkEmpty(text: String) -> Bool {
+        return text.count > 0
+    }
```
string isEmpty는어떨까요

hyejuuu commented on SsgSag/Screens/Review/ViewModels/SimpleReviewViewModel.swift:
```diff
@@ -0,0 +1,41 @@
+    func bind() {
+        Observable.combineLatest(oneLineObservable, advantageObservable, disadvantageObservable, honeyObservable, resultSelector: {
+            return self.checkEmpty(text: $0) && self.checkMinmumText(text: $1) && self.checkMinmumText(text: $2) && self.checkMinmumText(text: $3)
+        })
```
한꺼번에 많은 함수, 튜플을 처리하여 가독성이 떨어져 보입니다 :)

namsoo5 replied:
4개의 Observable가 작동할때마다 text를 검사해서 다음버튼을 활성화 해야하는데
가독성을 높이는 방법에는 뭐가 있을까요??

hyejuuu commented on SsgSag/Screens/Review/ViewModels/StarRatingViewModel.swift:
```diff
@@ -0,0 +1,34 @@
+    func bind() {
+        Observable
+            .zip(funDegreeObservable, proDegreeObservable, hardDegreeObservable, friendDegreeObservable, resultSelector: {$0+$1+$2+$3 >= 0})
+            .bind(onNext: { [weak self] bool in
```
마찬가지로 많은 옵저버블을 한번에 다루어 어떤 동작을 원하는지 잘 모르겠습니다 ㅠ

hyejuuu commented on SsgSag/Screens/Review/Views/Cell/ClubPhotoCollectionViewCell.swift:
```diff
@@ -7,22 +7,16 @@
     let view = UIView()
     let label = UILabel()
     var imgString: String! {
         willSet {
```
뷰 모델에 있어야할 프로퍼티로 보입니다

hyejuuu commented on SsgSag/Screens/Review/Views/Cell/InputCategoryCollectionViewCell.swift:
```diff
@@ -0,0 +1,35 @@
+    }
+    
+    func bind(model: BehaviorRelay<[CategorySelectModel]>, item: Int) {
+        categoryLabel.text = model.value[item].title
```
vm이 왜 BehaviorRelay가 되어야하는지 궁금합니다

namsoo5 replied:
테이블뷰를 만드는 부분에서 bind할때 모델형식이 BehaviorRelay라서 그대로 썻던것같습니다.!!
[CategorySelectModel]으로 수정했습니다.

hyejuuu commented on SsgSag/Screens/Review/Views/Cell/ReviewSearchTableViewCell.swift:
```diff
@@ -21,4 +33,51 @@ class ReviewSearchTableViewCell: UITableViewCell {
+        
+        labels.compactMap {
+            let lable = CategoryView(text: $0)
+            return lable
```
label 오타가 있군요

hyejuuu commented on SsgSag/Screens/Review/Views/Cell/ReviewSearchTableViewCell.swift:
```diff
@@ -21,4 +33,51 @@ class ReviewSearchTableViewCell: UITableViewCell {
+        categoryStack.trailingAnchor.constraint(equalTo: self.trailingAnchor, constant: -24).isActive = true
+        
+        labels.compactMap {
+            let lable = CategoryView(text: $0)
```
왜 compactMap을 사용하셨는지 궁금합니다

namsoo5 replied:
string으로 label을 만들고난뒤에 받은 배열을 다시 foreach로접근해서 스텍뷰에 넣으려고 생각해서썻는데
이제보니 중복되어 보이네요!!
구분하지않고 한번에 foreach로 처리하였습니다

hyejuuu commented on SsgSag/Screens/Review/Views/Cell/ReviewSearchTableViewCell.swift:
```diff
@@ -21,4 +33,51 @@ class ReviewSearchTableViewCell: UITableViewCell {
+            categoryStack.addSubview($0)
+        }
+        self.layoutIfNeeded()
+    }
```
layoutIfNeeded를 호출해야하는 이유가 있나요?

namsoo5 replied:
UI가 업데이트되는 로직을 잘몰라서
업데이트가 잘안될수도 있다는생각에 호출했습니다.

namsoo5 replied:
코드상 제거했습니다.!!

hyejuuu commented on SsgSag/Screens/Review/Views/Cell/ReviewSearchTableViewCell.swift:
```diff
@@ -21,4 +33,51 @@ class ReviewSearchTableViewCell: UITableViewCell {
+    
+    func ratePaint(score: Float) {
+        let stackStar = self.starStackView.subviews as! [UIImageView]
+        var score = score
```
강제 캐스팅은 위험해보입니다

hyejuuu commented on SsgSag/Screens/Review/Views/Cell/ReviewSearchTableViewCell.swift:
```diff
@@ -21,4 +33,51 @@ class ReviewSearchTableViewCell: UITableViewCell {
+            score -= 1
+            if score >= 0 {
+                $0.image = self.fullStar
+            } else if score > -1 {
```
self.fullStar 등 이미지를 이쪽에서 사용하게 된다면 lazy를 사용하는것도 좋을것 같습니다. 또한 ratePoint 함수는 테스팅이 가능하도록 구현할 수 있게 좀 더 분리하는게 어떨까싶네요

namsoo5 replied:
lazy적용하였고
ratePaint(score: Float, starStackView: UIStackView)로 수정하여 스텍뷰와 점수를 이용하여 테스팅가능하도록 해봤습니다

hyejuuu commented on SsgSag/Screens/Review/Views/ClubActInfoAlertViewController.swift:
```diff
@@ -0,0 +1,115 @@
+    func appearAnim(type: InputType) {
+        if type == .location {
+            self.locationView.isHidden = false
+           UIView.animate(withDuration: animationDuration, delay: 0, options: .curveEaseOut, animations: {
```
vm을 사용하여 구현해볼 수 있을것 같습니다

hyejuuu commented on SsgSag/Screens/Review/Views/ClubActInfoAlertViewController.swift:
```diff
@@ -0,0 +1,115 @@
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "ActLocationCell", for: indexPath) as! ActLocationTableViewCell
+        cell.locationLabel.text = self.locations[indexPath.row]
```
강제 캐스팅, safe없이 인덱스 접근은 위험할 수 있습니다

hyejuuu commented on SsgSag/Screens/Review/Views/ClubDetailViewController.swift:
```diff
@@ -101,7 +105,7 @@ class ClubDetailViewController: UIViewController {
     func requestClubInfo() {
         DispatchQueue.global().async {
-            ClubService.shared.requestClubInfo(clubIdx: self.clubIdx) { data in
+            ClubService().requestClubInfo(clubIdx: self.clubIdx) { data in
```
여기서 글로벌큐가 사용되야하는 이유가 뭘까요?

namsoo5 replied:
필요없을것같습니다.!! 
제거했습니다

hyejuuu commented on SsgSag/Screens/Review/Views/ClubInfoViewController.swift:
```diff
@@ -30,31 +31,30 @@ class ClubInfoViewController: UIViewController {
-            DispatchQueue.main.async {
+            .filter { $0 != nil }
+            .map { $0 }
+            .observeOn(MainScheduler.instance)
```
compactMap 을 사용해볼 수 있지않을까요?

hyejuuu commented on SsgSag/Screens/Review/Views/ClubInfoViewController.swift:
```diff
@@ -30,31 +31,30 @@ class ClubInfoViewController: UIViewController {
+            .observeOn(MainScheduler.instance)
+            .subscribe(onNext: { [weak self] data in
+                guard data != nil else { return }
+                self?.activeNumLabel.text = data!.activeNum
```
guard let data = data .. 를 사용하시면 강제 추출을 하지않으셔도됩니다

hyejuuu commented on SsgSag/Screens/Review/Views/ClubListViewController.swift:
```diff
@@ -24,10 +24,16 @@ class ClubListViewController: UIViewController {
-        ClubService.shared.requestClubList(curPage: curPage) { data in
-            guard data != nil else { return }
-            data!.forEach { self.cellData.append($0) }
+        ClubService().requestClubList(curPage: curPage) { data in
```
바로 인스턴스로 만들어 사용하게되면 해당 뷰컨트롤러를 테스팅하기 어렵습니다
프로퍼티로 가지고 있되 프로토콜 타입으로 선언해서 사용하는 것이 좋습니다

hyejuuu commented on SsgSag/Screens/Review/Views/Extension/ClubInfoVC+Extension.swift:
```diff
@@ -11,19 +11,19 @@ import UIKit
-        if self.infoPhotoURLSet.count > 6 {
+        if self.imgSet.count > 6 {
             return 6
         }
```
6이란 숫자는 어떤 의미를 가지나요?

hyejuuu commented on SsgSag/Screens/Review/Views/Extension/ClubInfoVC+Extension.swift:
```diff
@@ -11,19 +11,19 @@ import UIKit
         let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "ClubPhotoCell", for: indexPath) as! ClubPhotoCollectionViewCell
+    
+        cell.imgString = self.imgSet[indexPath.item]
         
```
vm이 할수있는일처럼 보이네요

hyejuuu commented on SsgSag/Screens/Review/Views/Extension/ClubInfoVC+Extension.swift:
```diff
@@ -11,19 +11,19 @@ import UIKit
+        let dataNum = self.imgSet.count
         if indexPath.item == 5 && dataNum > 6 {
             
             let gap = dataNum - 6
```
각 숫자가 무슨 의미를 가지는지 잘 모르겠습니다

hyejuuu commented on SsgSag/Screens/Review/Views/Extension/ReviewMainVC+Extension.swift:
```diff
@@ -36,22 +36,20 @@ extension ReviewMainViewController: UICollectionViewDataSource, UICollectionView
-        guard indexPath.item != 0 else { return }
-        self.simplerAlert(title: "아직 준비중이에요!!")
+        let cell = collectionView.cellForItem(at: indexPath) as! ReviewPageCollectionViewCell
+        self.tabTitle[indexPath.item].onFocus = true
```
강제 캐스팅은 조심해주세요

hyejuuu commented on SsgSag/Screens/Review/Views/FoundClubViewController.swift:
```diff
@@ -0,0 +1,99 @@
+    func bind(model: ClubActInfoModel) {
+        model.startDate
+            .distinctUntilChanged()
+            .subscribe(onNext: { [weak self] dateString in
```
distinctUntilChanged() 를 사용하실때 어떤 효과가 있을지 생각해보시는 것도 좋을듯합니다

hyejuuu commented on SsgSag/Screens/Review/Views/FoundClubViewController.swift:
```diff
@@ -0,0 +1,99 @@
+    }
+    
+    func isEnableButton() {
+        guard !(startDateLabel.text?.isEmpty ?? true) else {
```
vm이 가져야할 함수처럼 보입니다

hyejuuu commented on SsgSag/Screens/Review/Views/FoundClubViewController.swift:
```diff
@@ -0,0 +1,99 @@
+    
+    @IBAction func startDateClick(_ sender: Any) {
+        let nextVC = self.storyboard?.instantiateViewController(withIdentifier: "ClubActInfoAlertVC") as! ClubActInfoAlertViewController
+        clubactInfo.inputType = .start
```
강제캐스팅은 크래쉬를 일으킬수있습니다

hyejuuu commented on SsgSag/Screens/Review/Views/FoundClubViewController.swift:
```diff
@@ -0,0 +1,99 @@
+        nextVC.clubactInfo = self.clubactInfo
+        
+        present(nextVC, animated: true, completion: nil)
+    }
```
self를 쓰는 곳도 안쓰는 곳도 있어 혼란스러울수있겠네요

hyejuuu commented on SsgSag/Screens/Review/Views/InputUserClubInfoViewController.swift:
```diff
@@ -0,0 +1,182 @@
+            self.univOrLoaclImgView.isHidden = false
+        }
+    }
+    
```
vm에 있어야할 함수로 보입니다

hyejuuu commented on SsgSag/Screens/Review/Views/InputUserClubInfoViewController.swift:
```diff
@@ -0,0 +1,182 @@
+    }
+    
+    func isEnableButton() {
+        guard !(clubNameTextField.text?.isEmpty ?? true) else {
```
마찬가지로 vm에 위치하고 테스터블하게 구현하면 좋을듯합니다

hyejuuu commented on SsgSag/Screens/Review/Views/NotFoundClubViewController.swift:
```diff
@@ -0,0 +1,95 @@
+                } else if selectCount < 3 {
+                    self?.cellModel.value[index.item].onClick = true
+                    self?.selectCount += 1
+                }
```
vm에 정리되어 들어가야할 함수로 보이네요

hyejuuu commented on SsgSag/Screens/Review/Views/ReviewSearchViewController.swift:
```diff
@@ -7,15 +7,75 @@
+            
+          
+        viewModel.isEmpty
+            .asDriver()
```
어떤것을 위한 isEmpty인지 잘 모르겠습니다

hyejuuu commented on SsgSag/Screens/Review/Views/ReviewSearchViewController.swift:
```diff
@@ -7,15 +7,75 @@
+        viewModel.isEmpty
+            .asDriver()
+            .drive(onNext: { [weak self] bool in
+                guard let title = self?.searchTexfield.text else {return}
```
bool 보다는 가독성이 좋은 다른 변수명을 택해주세요

hyejuuu commented on SsgSag/Screens/Review/Views/ReviewSearchViewController.swift:
```diff
@@ -7,15 +7,75 @@
+                self?.searchLabel.isHidden = !bool
+                self?.clubRegisterButton.isHidden = !bool
+            })
+            .disposed(by: disposeBag)
```
전부 vm에서 처리 가능하게 보이네요

hyejuuu commented on SsgSag/Screens/Review/Views/SimpleReviewViewController.swift:
```diff
@@ -0,0 +1,152 @@
+            .drive(onNext: { [weak self] text in
+               
+                self?.advantageBlackLabel.isHidden = text.count == 0 ? false : true
+                self?.simpleReviewViewModel.advantageObservable.accept(text)
```
vm에 들어가야하지 않을까요?

hyejuuu commented on SsgSag/Screens/Review/Views/SimpleReviewViewController.swift:
```diff
@@ -0,0 +1,152 @@
+            self?.disadvantageBlackLabel.isHidden = text.count == 0 ? false : true
+            self?.simpleReviewViewModel.disadvantageObservable.accept(text)
+            
+        })
```
마찬가지 입니다

hyejuuu commented on SsgSag/Screens/Review/Views/SimpleReviewViewController.swift:
```diff
@@ -0,0 +1,152 @@
+        
+        let updateText = curString.replacingCharacters(in: stringRange, with: string)
+        return updateText.count < 20
+    }
```
20은 어떤 의미를 갖나요?

hyejuuu commented on SsgSag/Screens/Review/Views/SimpleReviewViewController.swift:
```diff
@@ -0,0 +1,152 @@
+        
+        guard let curString = textField.text else { return false }
+        guard let stringRange = Range(range, in: curString) else { return false }
+        
```
위 와 같은 로직이 textField 델리게이트 메서드 내부로 들어가면서 테스팅이 어렵게 되었습니다

hyejuuu commented on SsgSag/Screens/Review/Views/StarRatingViewController.swift:
```diff
@@ -0,0 +1,212 @@
+    var clubactInfo: ClubActInfoModel!
+    let starRatingModel = StarRatingViewModel()
+    let disposeBag = DisposeBag()
+    
```
vm에서 이미지 네임만 들고 있는것도 방법입니다

hyejuuu commented on SsgSag/Screens/Review/Views/StarRatingViewController.swift:
```diff
@@ -0,0 +1,212 @@
+            self.scrollView.transform = CGAffineTransform(scaleX: 1, y: 0)
+        }
+        self.view.layoutIfNeeded()
+        
```
layoutIfNeeded()를 호출하는 이유가있나요?

namsoo5 replied:
삭제했습니다.!

hyejuuu commented on SsgSag/Screens/Review/Views/StarRatingViewController.swift:
```diff
@@ -0,0 +1,212 @@
+                            self?.scrollAppearAnim()
+                            self?.scrollAppear = true
+                        }
+                    }
```
이부분을 풀어볼 방법이 없을까요?

namsoo5 replied:
`
if !(self?.scrollAppear ?? false) {
    self?.scrollAppearAnim()
}
`
위처럼 수정하고 scrollAppearAnim 함수안에 scrollAppear 변수가 truer가 되도록수정했는데
의도하신것 처럼 되었는지 모르겠습니다...

hyejuuu commented on SsgSag/Screens/Review/Views/StarRatingViewController.swift:
```diff
@@ -0,0 +1,212 @@
+        .disposed(by: disposeBag)
+        
+        let friendStarButtons = friendStack.subviews as! [UIButton]
+        viewModel
```
강제 캐스팅은 위험하니 조심해주세요

namsoo5 replied:
if let 사용으로 수정하였습니다!!
모든경우를 실행해야하므로 실행이 안될시 아래있는 코드를 실행 못 할 수도 있는 guard let을 지양했습니다.

hyejuuu commented on SsgSag/Screens/Review/Views/StarRatingViewController.swift:
```diff
@@ -0,0 +1,212 @@
+        buttons = friendStack.subviews as! [UIButton]
+        buttons.forEach {$0.addTarget(self, action: #selector(friendClick(sender:)), for: .touchUpInside)}
+    }
+    
```
위 부분 전체적으로 수정이 필요해보입니다

namsoo5 replied:
아래와 동일합니다
계속 생각해봤는데 어떻게 처리 로직을 짜야할지 잘 모르겠습니다...

hyejuuu commented on SsgSag/Screens/Review/Views/StarRatingViewController.swift:
```diff
@@ -0,0 +1,212 @@
+    
+    @objc func recommendClick(sender: UIButton) {
+        starRatingModel.recommendDegreeObservable.accept(sender.tag)
+    }
```
아래 로직들도 버튼이라면 rx로 처리가 가능해보입니다

namsoo5 replied:
계속 생각해봤는데 어떻게 처리 로직을 짜야할지 잘 모르겠습니다...

namsoo5 committed:
UPDATE: 리뷰를 토대로 부분적 리팩토링, 동아리 정보, 후기없을때 뜨는 뷰 구현

- 부족한 부분 코드수정
- 후기 없을시 뜨는 뷰 구현

namsoo5 committed:
UPDATE: 동아리 등록 첫번째 Step 뷰구현

- 등록뷰 구현
- 클릭시 뜨는 지역, 카테고리 고르는 VC생성

hyejuuu commented:
양이 많으니 조금 끊어서 볼께요 한번뵙고하는게 빠를수도있겠어요 ㅎㅎ

namsoo5 committed:
UPDATE: 동아리 등록 - 카테고리 선택, 지역 선택뷰 구현

- 버튼클릭시 해당하는 창띄우기
- 카테고리 선택시 선택한 카테고리 Cell추가

namsoo5 committed:
UPDATE: 동아리등록 두번째 Step 구현

namsoo5 committed:
UPDATE: 동아리 미관계자 등록시 통신구현

namsoo5 committed:
UPDATE: 동아리 등록 통신 구현

- 동아리 관계자일때
- 동아리 후기등록시 관계자일때

namsoo5 committed:
UPDATE: 화면터치시 키보드숨기기

namsoo5 committed:
UPDATE: 메인 디자인 수정, 상세정보 통신값 적용

- 통신값 바인딩
- 메인 디자인 수정
- QA사항 적용
- textfield pedding적용

namsoo5 committed:
UPDATE: 연합동아리 자동완성 구현 및 UX구현

- 스크롤시 자동완성 사라짐
- 왼쪽모서리에서 오른쪽으로 슬라이드시 종료기능구현

namsoo5 committed:
UPDATE: 학교이름 자동완성을 통한 동아리이름 자동완성구현

namsoo5 committed:
UPDATE: 동아리 후기등록 통신구현

- 플로우에 맞도록 수정
- 등록되지않은 동아리 입력 통신구현

namsoo5 committed:
UPDATE: 동아리입력 카테고리선택 셀 왼쪽정렬구현

- 카테고리 셀 왼쪽정렬

namsoo5 committed:
UPDATE: 동아리 detail뷰 통신데이터 삽입

- 뷰에 통신값 적용

namsoo5 committed:
UPDATE: 카테고리 통신을통한 셀 바인딩

- 셀방향 버그수정
- 카테고리 스트링 -> 배열로 변환

namsoo5 committed:
FIX: 카테고리 CollectionView 버그수정

- 메인리스트 카테고리수정
- 동아리 디테일 카테고리수정

namsoo5 committed:
UPDATE: 별점평가 기준 팝업 구현

- 평점 기준 팝업구현

namsoo5 committed:
UPDATE: 블로그 등록뷰 구현

- 블로그 등록 플로우 구현
- 후기탭 정보,후기 더블탭 간격조정

namsoo5 committed:
UPDATE: 슥삭 후기 더보기 구현

- 디자인 수정사항 반영: 폰트사이즈, 색 조정

namsoo5 committed:
FIX: 후기쓰기 버그수정

- 동아리이름이 안넘어가던 버그 수정
- 터치시 키보드내리기

namsoo5 committed:
UPDATE: 메인 팝업창 구현

namsoo5 committed:
UPDATE: 후기 더보기 도움돼요 버튼기능구현

namsoo5 committed:
UPDATE: 후기더보기 도움돼요 취소버튼 기능구현

namsoo5 committed:
UPDATE: 동아리 상세 도움돼요 버튼 기능 구현

namsoo5 committed:
FIX: 동아리상세 정보/후기탭 후기에서 더보기시 버그 수정

- 높이에 맞도록 셀크기 재조정

namsoo5 committed:
UPDATE: 메인 팝업 이미지변경, 동아리등록시 조건 미충족 알림 구현

- 20자이상 입력 팝업 구현
- 메인 이미지 변경

namsoo5 committed:
UPDATE: 후기수정 뷰디자인 구현

namsoo5 committed:
UPDATE: 후기 수정하기 동아리활동정보 기능구현

- 동아리활동정보
- 디자인QA적용

namsoo5 committed:
UPDATE: Asset파일 교체

- 쓰지않는 더미에셋 삭제
- 새로운 탭바아이템 추가

namsoo5 committed:
FIX: TabBar Item 워딩추가에 따른 높이조절

namsoo5 committed:
UPDATE: 후기수정 별점, 간단평가 기능구현

- 별점기능 구현
- 평가기능 구현

namsoo5 committed:
UPDATE: 기본 리뷰데이터 받아오기

- 데이터받아와서 초기값 구현

namsoo5 committed:
UPDATE: 후기수정 불필요코드 삭제

namsoo5 committed:
UPDATE: 후기수정시 기본값 불러오는 도중 데이터형식 수정

- yyyy-mm 형식의 데이터입력에서 yyyy년 M월 형식으로 변경

namsoo5 committed:
UPDATE: 별점평가 애니메이션 수정

- 중앙 기준 transform을 위에서 아래로 펼쳐지도록 애니메이션구현

namsoo5 committed:
UPDATE: 동아리 상세 정보/후기탭 후기Cell터치시 더보기 기능 수정

- 기존의 더보기 버튼클릭시 확장이아닌 셀클릭시 확장

namsoo5 committed:
UPDATE: 동아리상세 정보/후기탭 후기 더보기시에도 확장되도록 구현

namsoo5 committed:
FIX: 메인 동아리리스트 교내/연합 구분 및 버그수정

- 교내 동아리, 연합동아리 구분
- 기존 collectionView 이벤트를 Rx로 변경하여 구현

namsoo5 committed:
UPDATE: 동아리 상세 정보/후기 후기탭 수정

- 블로그, 슥삭후기밑에 선추가
- 후기 개수 추가

namsoo5 committed:
UPDATE: 디폴트값 제거

- 서버로딩이 느릴시 보이는 디폴트제거

namsoo5 committed:
UPDATE: 업데이트전 탭바 로그인시 가능하도록 수정, 팝업로직수정, 통신버튼 중복수정

- 별점평가 스크롤버그수정
- 제출버튼
- 탭바 로그인시 사용가능하도록
- 후기제출, 동아리등록버튼 중복터치 제거
